### PR TITLE
[SYCL][Matrix] Add new joint matrix examples: one for half type and one for query

### PR DESCRIPTION
**Pull request by dkhaldi**
This PR adds one test for half type test that was missing. Note that half is only supported on DPAS (not on AMX).
The other test shows a query default use case.

Signed-off-by: Dounia Khaldi <dounia.khaldi@intel.com>

### CODE REVIEW

File: SYCL/Matrix/joint_matrix_half.cpp [hunk 1]
```diff
@@ -0,0 +1,165 @@
+//==-------- joint_matrix_half.cpp  - DPC++ joint_matrix------------ ----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: matrix
```
Comment (vladimirlaz) [inline SYCL/Matrix/joint_matrix_half.cpp:8]: The tests are effectively disabled due to this rule:
```
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_bf16.cpp (446 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_half.cpp (447 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_query_default.cpp (448 of 648)
```
Do we need to enable it by default in CI or it is not supported on most of the HW?

Comment (dkhaldi) [reply SYCL/Matrix/joint_matrix_half.cpp:8]: @yuxianch, can you please confirm the tests are running?
In our last discussion we already went over this to make sure matrix is implementated and the tests are running on DPAS and SDE

Comment (yuxianch) [reply SYCL/Matrix/joint_matrix_half.cpp:8]: Yes, the tests are running.

Comment (vladimirlaz) [reply SYCL/Matrix/joint_matrix_half.cpp:8]: Looking into the tests in the PR I see that all tests in the Matrix folder are skipped (marked UNSUPPORTED):
```
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_bf16.cpp (446 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_half.cpp (447 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_query_default.cpp (448 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_ss_int8.cpp (449 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_su_int8.cpp (450 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_us_int8.cpp (451 of 648)
[2021-10-12T21:33:13.996Z] UNSUPPORTED: SYCL :: Matrix/joint_matrix_uu_int8.cpp (452 of 648)
```

**Review by yuxianch — APPROVED** [review]
LGTM for these REQUIRES/RUN commands.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**